src/hb-ot-var-common.hh

```diff
@@ -528,6 +584,326 @@ struct tuple_delta_t
 
     return out;
   }
+
+  /* deltas should be compiled already before we compile tuple
+   * variation header cause we need to fill in the size of the
+   * serialized data for this tuple variation */
+  //TODO(qxliu):add option to use sharedTuples in gvar
+  bool compile_tuple_var_header (const hb_map_t& axes_index_map,
+                                 unsigned points_data_length,
+                                 const hb_map_t& axes_old_index_tag_map)
+  {
+    if (!compiled_deltas) return false;
+
+    unsigned cur_axis_count = axes_index_map.get_population ();
+    /* allocate enough memory: 1 peak + 2 intermediate coords + fixed header size */
+    unsigned alloc_len = 3 * cur_axis_count * (F2DOT14::static_size) + 4;
+    char *p = (char *) hb_calloc (alloc_len, sizeof (char));
```

Comment:
Likewise here, you could use a vector_t<char> + a call to resize() to allocate the memory. That way the vector would own the memory. At the end you would move it to compiled_tuple_header. This would then ensure memory is automatically cleaned up so there's no need to call hb_free(p) before each return.

Same in the other compile functions that allocate memory.